src/main/java/io/hops/util/CredentialsProvider.java

```diff
@@ -0,0 +1,190 @@
+
+/**
+ * AWS temporary credential provider.
+ */
+public class CredentialsProvider {
+  private static final Logger LOG = Logger.getLogger(CredentialsProvider.class.getName());
+  
+  private CredentialsProvider() {
+  }
+  
+  /**
+   * Get temporary credentials and set spark context hadoop configuration
+   * @param role
+   * @return Credentials
+   * @throws CloudCredentialException
+   */
+  public static Credentials assumeRole(String role) throws CloudCredentialException {
+    return assumeRole(role, 3600);
+  }
+  
+  /**
+   * Get temporary credentials and set spark context hadoop configuration
+   * @param role
+   * @param durationSeconds
+   * @return Credentials
+   * @throws CloudCredentialException
+   */
+  public static Credentials assumeRole(String role, int durationSeconds) throws CloudCredentialException {
+    String uniqueID = UUID.randomUUID().toString();
+    return assumeRole(role, "session_" + uniqueID, durationSeconds);
+  }
+  
+  /**
+   * Get temporary credentials and set spark context hadoop configuration and system properties.
+   * @param role
+   * @param roleSessionName
+   * @param durationSeconds
+   * @return Credentials
+   * @throws CloudCredentialException
+   */
+  public static Credentials assumeRole(String role, String roleSessionName, int durationSeconds)
+    throws CloudCredentialException {
+    Response response;
+    try {
+      HashMap<String, Object> queryParams = new HashMap<>();
+      queryParams.put(Constants.HOPSWORKS_CLOUD_SESSION_TOKEN_RESOURCE_QUERY_ROLE, role);
+      queryParams.put(Constants.HOPSWORKS_CLOUD_SESSION_TOKEN_RESOURCE_QUERY_SESSION, roleSessionName);
+      queryParams.put(Constants.HOPSWORKS_CLOUD_SESSION_TOKEN_RESOURCE_QUERY_SESSION_DURATION, durationSeconds);
+      response = Hops.clientWrapper(
+          "/" + Constants.HOPSWORKS_CLOUD_RESOURCE + "/" + Constants.HOPSWORKS_CLOUD_SESSION_TOKEN_RESOURCE,
+          HttpMethod.GET, queryParams);
+    } catch (HTTPSClientInitializationException | JWTNotFoundException e) {
+      throw new CloudCredentialException(e.getMessage());
+    }
+    LOG.log(Level.INFO, "******* response.getStatusInfo():" + response.getStatusInfo());
+    JSONObject jsonObject = getResponse(response);
+    Credentials credentials = getCredentialsFromJson(jsonObject);
+    setSparkHadoopConf(credentials);
+    setSystemProperties(credentials);
+    return credentials;
+  }
+  
+  /**
+   * Get all roles mapped to the current project
+   * @return list of roles
+   * @throws CloudCredentialException
+   */
+  public static String[] getRoles() throws CloudCredentialException {
+    JSONObject roles = getCloudRoles(null);
+    JSONArray items =  roles.getJSONArray(Constants.JSON_ARRAY_ITEMS);
+    String[] cloudRoles = new String[items.length()];
+    for (int i = 0; i < items.length(); i++) {
+      JSONObject role = (JSONObject) items.get(i);
+      cloudRoles[i] = (String) role.get(Constants.JSON_CLOUD_ROLE);
+    }
```

Comment:
That looks like hard to maintain if the DTO returned by the REST API changes over time. Also is not easy to read.
We can use JAXB to do it, like https://github.com/logicalclocks/hops-util/blob/master/src/main/java/io/hops/util/featurestore/FeaturestoreHelper.java#L143 

and later on we can move to the codebase to using something like https://github.com/logicalclocks/feature-store-api/blob/master/java/src/main/java/com/logicalclocks/hsfs/metadata/HopsworksClient.java#L134

Reply:
JAXB does not work I tried.  And if the dto changes we still need to change the dto here so It is all the same.

Reply:
Ok let's create a JIRA to fix it after we migrate to hsfs